scripts/utils/printing_tools.js

```diff
@@ -0,0 +1,52 @@
+const BN = require("bn.js")
+const bnOne = new BN(1)
+const bnTwo = new BN(2)
+const bnTen = new BN(10)
+const bn256 = new BN(256)
+const bnMaxUint = bnTwo.pow(bn256).sub(bnOne)
+
+/**
+ * A generalized version of "toWei" for tokens with an arbitrary amount of digits.
+ * If the decimal representation has more digits than the maximum amount possible, then the extra digits are truncated.
+ * @param {string} amount User-friendly representation for the amount of some ERC20 token
+ * @param {integer} digits Maximum number of decimal digits of the token
+ * @return {string} decimal representation of the number of token units corresponding to the input amount
+ */
+const fromUserToMachineReadable = function (amount, digits) {
+  if (digits < 0 || digits > 256) throw Error("Invalid number of digits for ERC20 token: " + digits.toString())
+  const re = /^(\d+)(\.(\d+))?$/ // a sequence of at least one digit (0-9), followed by optionally a dot and another sequence of at least one digit
```

Comment:
Many numbers get converted automatically into an e representation 1e-4 for example